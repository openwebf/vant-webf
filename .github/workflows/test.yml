name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - dev

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
