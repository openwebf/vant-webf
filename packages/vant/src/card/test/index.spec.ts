import { Card } from '..';
import { mount } from '../../../test';

test('should emit click event after clicked', () => {
  const onClick = jest.fn();
  const wrapper = mount(Card, {
    props: {
      onClick,
    },
  });
  wrapper.trigger('click');
  expect(onClick).toHaveBeenCalledTimes(1);
});

test('should emit click-thumb event after clicking thumb', () => {
  const wrapper = mount(Card, {
    props: {
      thumb: 'xx',
    },
  });

  wrapper.find('.van-card__thumb').trigger('click');
  expect(wrapper.emitted('click-thumb')!).toHaveLength(1);
});

test('should render price and num slot correctly', () => {
  const wrapper = mount(Card, {
    slots: {
      num: () => 'Custom Num',
      price: () => 'Custom Price',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render origin-price slot correctly', () => {
  const wrapper = mount(Card, {
    slots: {
      'origin-price': () => 'Custom Origin Price',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render bottom slot correctly', () => {
  const wrapper = mount(Card, {
    slots: {
      bottom: () => 'Custom Bottom',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('render thumb and tag slot correctly', () => {
  const wrapper = mount(Card, {
    slots: {
      tag: () => 'Custom Tag',
      thumb: () => 'Custom Thumb',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render title and desc slot correctly', () => {
  const wrapper = mount(Card, {
    slots: {
      title: () => 'Custom Title',
      desc: () => 'Custom desc',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render price and price-top slot correctly', () => {
  const wrapper = mount(Card, {
    slots: {
      price: () => 'Custom Price',
      'price-top': () => 'Custom Price-top',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
