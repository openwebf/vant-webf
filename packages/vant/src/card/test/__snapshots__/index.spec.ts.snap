// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render thumb and tag slot correctly 1`] = `
<div class="van-card">
  <div class="van-card__header">
    <a class="van-card__thumb">
      Custom Thumb
      <div class="van-card__tag">
        Custom Tag
      </div>
    </a>
    <div class="van-card__content">
      <div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render bottom slot correctly 1`] = `
<div class="van-card">
  <div class="van-card__header">
    <div class="van-card__content">
      <div>
      </div>
      <div class="van-card__bottom">
        Custom Bottom
      </div>
    </div>
  </div>
</div>
`;

exports[`should render origin-price slot correctly 1`] = `
<div class="van-card">
  <div class="van-card__header">
    <div class="van-card__content">
      <div>
      </div>
      <div class="van-card__bottom">
        <div class="van-card__origin-price">
          Custom Origin Price
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render price and num slot correctly 1`] = `
<div class="van-card">
  <div class="van-card__header">
    <div class="van-card__content">
      <div>
      </div>
      <div class="van-card__bottom">
        <div class="van-card__price">
          Custom Price
        </div>
        <div class="van-card__num">
          Custom Num
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render price and price-top slot correctly 1`] = `
<div class="van-card">
  <div class="van-card__header">
    <div class="van-card__content">
      <div>
      </div>
      <div class="van-card__bottom">
        Custom Price-top
        <div class="van-card__price">
          Custom Price
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render title and desc slot correctly 1`] = `
<div class="van-card">
  <div class="van-card__header">
    <div class="van-card__content">
      <div>
        Custom TitleCustom desc
      </div>
    </div>
  </div>
</div>
`;
